buildSrc/src/main/java/io/micronaut/guides/core/BuildDiffLinkSubstitution.java

```diff
@@ -29,21 +28,20 @@ public String substitute(String str, GuidesOption option, Guide guide) {
                     .findFirst()
                     .orElse(null);
 
-            String link = "https://micronaut.io/launch?" +
-                    featureNames(line, app, option).stream()
-                            .map(f -> "features=" + f)
-                            .collect(Collectors.joining("&")) +
-                    "&lang=" + option.getLanguage().name() +
-                    "&build=" + option.getBuildTool().name() +
-                    "&test=" + option.getTestFramework().name() +
-                    "&name=" + (appName.equals(guidesConfiguration.getDefaultAppName()) ? "micronautguide" : appName) +
-                    "&type=" + app.applicationType().name() +
-                    "&package=example.micronaut" +
-                    "&activity=diff" +
-                    "[view the dependency and configuration changes from the specified features, window=\"_blank\"]";
-
-            String res = "NOTE: If you have an existing Micronaut application and want to add the functionality described here, you can " +
-                    link + " and apply those changes to your application.";
+            UriBuilder uriBuilder = UriBuilder.of(guidesConfiguration.getLauncherUrl())
+                    .queryParam("lang", option.getLanguage().name())
+                    .queryParam("build", option.getBuildTool().name())
+                    .queryParam("test", option.getTestFramework().name())
+                    .queryParam("name", appName.equals(guidesConfiguration.getDefaultAppName()) ? "micronautguide" : appName)
+                    .queryParam("type", app.applicationType().name())
+                    .queryParam("package", guidesConfiguration.getPackageName())
+                    .queryParam("activity", "diff");
+            featureNames(line, app, option).forEach(f -> uriBuilder.queryParam("features", f));
+
+            String res = "NOTE: If you have an existing Micronaut application and want to add the functionality described here, you can "
```

Comment:
I think his copy should be in a common snippet instead. I don't think we should have copy trapped in the java classes.